packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1136,7 +1136,13 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
       }
 
       final double snackBarYOffsetBase;
-      if (floatingActionButtonRect.size != Size.zero && isSnackBarFloating) {
+      final bool showAboveFab = currentFloatingActionButtonLocation != FloatingActionButtonLocation.startTop
```

Comment:
This should use a switch statement. Dart 3 makes it really quite nice!

For example:

```dart
final BorderRadius childrenGroupBorderRadius = switch (type) {
        CupertinoListSectionType.insetGrouped => _kDefaultInsetGroupedBorderRadius,
        CupertinoListSectionType.base => BorderRadius.zero,
      };
```

Reply:
Nice! I switched to Dart 3 switch :smile: